### PR TITLE
Add blog post about customization of host-deploy

### PR DESCRIPTION
**Pull request by machacekondra**
cc @mwperina

### CODE REVIEW

**Comment by machacekondra** [comment]
ping

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+comments: true
+published: true
+---
+
+In 4.2 release we have introduced a possibility to customize the host-deploy proccess by running the Ansible post-tasks after the host-deploy process successfully finishes.
+
+READMORE
+
+## The reason
+Prior to 4.2 release when user need to customize the IPTables configuration of the hosts, he had to do database hacking. In 4.2 we have introduced the support for FirewallD,
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: No database hacking, everything was exposed via engine-config

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+and we wanted to add users the ability to customize the FirewallD configuration as well. Most people didn't like the database modification, so we have introduced Ansible integration,
+which is much more easier and powerfull than the database modification.
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: Please replace database modification with engine-config

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+process for example to open some more FirewallD ports.
+
+Those additional tasks can be added to following file:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml
+```
+
+## Example
+An example post-tasks file is provided by `ovirt-engine-tools` RPM, at following location:
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: Why is it provided by ovirt-engine-tools? Tools don't have anything in common with Ansible support, no?

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+>     immediate: yes
+>     state: enabled
+>
+> - name: Print debug information
+>   debug:
+>     msg: "My custom tasks was successfully executed!"
+> EOF
+```
+
+Now add the new host, and see the Ansible host-deploy log, if our custom post-tasks was executed,
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: Install a new host or reinstall existing host

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+2017-12-08 15:36:12,633 p=4777 u=omachace |  changed: [192.168.121.36]
+2017-12-08 15:36:12,639 p=4777 u=omachace |  TASK [Print debug information] *************************************************
+2017-12-08 15:36:12,663 p=4777 u=omachace |  ok: [192.168.121.36] => {
+    "msg": "My custom tasks was successfully executed!"
+}
+2017-12-08 15:36:12,665 p=4777 u=omachace |  PLAY RECAP *********************************************************************
+2017-12-08 15:36:12,665 p=4777 u=omachace |  192.168.121.36             : ok=23   changed=4    unreachable=0    failed=0
+```
+
+As you can see we use same log naming as we used for the otopi based deploy process but we have added `-ansible` suffix to the log file name.
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: I'd also mention that post-host-deploy task can be used for any task, not only firewalld customization. And we should also mention that IPTables support is deprecated in 4.2 and will be removed in 4.3

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 1 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 2 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 3 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: We provide there all the playbooks and *sh scripts:

 https://github.com/oVirt/ovirt-engine/blob/master/ovirt-engine.spec.in#L1121

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 4 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 5 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 3 above)
Comment (mwperina) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: OK, I somehow missed that during reviews, `ovirt-engine-tools` package should provide only command line tools. But I'd rather not mention the package name directly (if we move playbooks into different package), I'd just mention that this playbook is provided as a part of ovirt-engine installation

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+## Example
+An example post-tasks file is provided by `ovirt-engine-tools` RPM, at following location:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml.example
+```
+
+But to see this whole thing in action, you need to create proper file without the `.example` suffix and proper custom tasks:
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: This is just and example file, to add some task into host deploy flow, you need to create below mentioned file and add some proper Ansible custom tasks:

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+>     immediate: yes
+>     state: enabled
+>
+> - name: Print debug information
+>   debug:
+>     msg: "My custom tasks was successfully executed!"
+> EOF
+```
+
+Install a new host or reinstall existing host, and see the Ansible host-deploy log, if our custom post-tasks was executed,
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: "... our custom ..." -> " ... your custom ..."

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+you can check it by seeing the following log file:
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: Here I'd rather mention that relevant log file is shown in Events view for relevant host Installation or Reinstallation. Showing below file may be confusing.

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+2017-12-08 15:36:12,639 p=4777 u=omachace |  TASK [Print debug information] *************************************************
+2017-12-08 15:36:12,663 p=4777 u=omachace |  ok: [192.168.121.36] => {
+    "msg": "My custom tasks was successfully executed!"
+}
+2017-12-08 15:36:12,665 p=4777 u=omachace |  PLAY RECAP *********************************************************************
+2017-12-08 15:36:12,665 p=4777 u=omachace |  192.168.121.36             : ok=23   changed=4    unreachable=0    failed=0
+```
+
+As you can see we use same log naming as we used for the otopi based deploy process but we have added `-ansible` suffix to the log file name.
+Note that IPTables are deprecated in version 4.2 and will be removed in version 4.3
```
Comment (mwperina) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: Note that host's firewall type `iptables` is deprecated in version 4.2 and will be removed in version 4.3

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+---
+
+In 4.2 release we have introduced a possibility to customize the host-deploy proccess by running the Ansible post-tasks after the host-deploy process successfully finishes.
+
+READMORE
+
+## The reason
+Prior to 4.2 release when user need to customize the IPTables configuration of the hosts, he had to change it via `engine-config`. In 4.2 we have introduced the support for FirewallD,
+and we wanted to add users the ability to customize the FirewallD configuration as well. Most people didn't like the database modification, so we have introduced Ansible integration,
+which is much more powerfull than using `engine-config`.
```
[the remark this answers is not in the document]
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+## The host-deploy playbook
```
[the remark this answers is not in the document]
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+Currently when the host is added to the engine or reinstalled, we execute the host-deploy playbook
+located at `/usr/share/ovirt-engine/playbooks/ovirt-host-deploy.yml`.
+
+This playbook executes few internal roles to configure for example OVN driver, libvirt guests or FirewallD.
+
+## Special tasks file
+
+As part of this role we also include additional tasks, which could be written by user, to modify the host-deploy
```
[the remark this answers is not in the document]
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 3 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 6 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 7 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

File: source/blog/2017-12-08-host-deploy-customization.html.md (hunk 8 above)
Comment (machacekondra) [reply source/blog/2017-12-08-host-deploy-customization.html.md]: done

**Comment by mwperina** [comment]
@JohnMarksRH

**Comment by mwperina** [comment]
@mykaul @sandrobonazzola

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+## The reason
+Prior to oVirt 4.2 release administrators could customize host's firewall rules using `engine-config` option `IPTablesConfigSiteCustom`.
+Unfortunately writing custom `iptables` rules into string value to be used in `engine-config` was very user unfriendly and using `engine-config` to provide custom
+`firewalld` rules would be even much worse. Because of above we have introduced Ansible integration as a part of host deploy flow, which allows administrators to
+add any custom actions executed on the host during host deploy flow.
+
+## Special tasks file
+
+As part of this role we also include additional tasks, which could be written by user, to modify the host-deploy
```
Comment (mykaul) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: user -> the user

File: source/blog/2017-12-08-host-deploy-customization.html.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+This post-task file is executed as part of host-deploy process just before setup network invocation.
+
+## Example
+An example post-tasks file is provided by ovirt-engine installation, at following location:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml.example
+```
+
+This is just and example file, to add some task into host deploy flow, you need to create below mentioned file and add some proper Ansible custom tasks:
```
Comment (mykaul) [inline source/blog/2017-12-08-host-deploy-customization.html.md]: and -> an

**Pull request merged**